include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -135,36 +135,81 @@ struct format_sam
 
     //!\brief The format version string.
     static constexpr char format_version[4] = "1.6";
-};
 
-} // namespace seqan3
+protected:
+    //!\privatesection
+    // Note that some member variables and functions are protected, because format_bam inherits from format_sam.
```

Comment:
Since these files are extra-ordinarily long, I would only function signatures into the class and define the functions below. Just so that someone looking at the class definition can more easily assess what the API is.

Reply:
(just a suggestion)

Reply:
done